### PR TITLE
Variogram fitting update

### PR DESCRIPTION
**Pull request by MuellerSeb**
This PR introduces an updated variogram fitting routine:
- fixing sill possible now
- `loss` is now selectable for smoother handling of outliers
- r2 score can now be returned to get an impression of the goodness of fitting
- weights can be passed
- instead of deselecting parameters, one can also give fix values for each parameter

Also:
- minor doc fixes
- skip travis builds for PRs from our own repo
- better setting of bounds in CovModel

### CODE REVIEW

**Comment by MuellerSeb** [comment]
@LSchueler : When you are ok with that, I would merge.

File: gstools/covmodel/fit.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+                    + "the set nugget should be less than the given sill."
+                )
+            para_select["var"] = False
+            model.var = sill - model.nugget
+        else:
+            # deselect the nugget, to recalculate it accordingly
+            # nugget = sill - var
+            para_select["nugget"] = False
+    else:
+        constrain_sill = False
```
Comment (LSchueler) [inline gstools/covmodel/fit.py:205]: Would it make sense to outsource stuff like this to helper functions for better readability?

File: gstools/covmodel/fit.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+    r2_score : :class:`float`, optional
+        r2 score of the curve fitting results. Only if return_r2 is True.
+
+    Notes
+    -----
+    You can set the bounds for each parameter by accessing
+    :any:`CovModel.set_arg_bounds`.
+
+    The fitted parameters will be instantly set in the model.
+    """
```
Comment (LSchueler) [inline gstools/covmodel/fit.py:148]: Great doc!

File: gstools/covmodel/fit.py (hunk 1 above)
Comment (MuellerSeb) [reply gstools/covmodel/fit.py:238]: Done. Thanks for the reminder!

File: gstools/covmodel/fit.py (hunk 2 above)
Comment (MuellerSeb) [reply gstools/covmodel/fit.py:148]: Thanks :smile_cat:

**Comment by MuellerSeb** [comment]
> The fit_variogram function is veeery long. If you promise to refactor it some time, then I'm fine with a merge right now ;-)

Refactored. You were totally right!

**Pull request merged**